adapter/internal/ga/client.go

```diff
@@ -323,3 +330,28 @@ func FetchAPIsFromGA() []*APIEvent {
 		time.Sleep(1 * time.Second)
 	}
 }
+
+// Related to issue: https://github.com/wso2-enterprise/choreo/issues/14460
+// This temporary function is added to prevent redeploying of all the existing APIs due to change done
+// in API revision ID to handle environment specific properties. New API revision ID will be in the format
+// of revisionId_timestamp and the currentGAAPI.RevisionUUID == api.RevisionUUID returns fails for all the
+// existing APIs. This function is added to prevent redeploying of APIs until a specific timestamp.
+// Remove this function after the GA is updated with revisionId_timestamp related changes.
+// 1. X -> return X
+// 2. X_t where t < T -> return X
+// 3. X_t where t >= T -> return X_t
+func preprocessRevisionID(revisionID string) string {
+	revisionIDSplitted := strings.Split(revisionID, "_")
+	if len(revisionIDSplitted) == 2 {
+		timestamp, err := strconv.Atoi(revisionIDSplitted[1])
+		if err != nil {
+			// Unreachable
+			logger.LoggerGA.Infof("Revision Id associated timestamp conversion to integer failed: %v\n", err)
```

Comment:
isn't this an error log?

Reply:
Since this whole PR anyway going to be reverted once the GA updates, I thought it is okay to have it as an info log.